[2.6.0] `./configure --without-docbook && make clean all` fails with `xmlwf.xml: command not found`

**Related**: https://github.com/libexpat/libexpat/issues/253

This happens when running the tests if `make check` is ran.

After a bisect, this is the offending change: https://github.com/libexpat/libexpat/commit/7384c88f9aa7dd49f22ae08f2a65c57b5dda2f9e

Reverting the change fixes the error. It seems to me the lack of a pre-renderered xmlwf.1 file sets WITH_DOCBOOK to true even if --without-docbook is specified (or if the detection doesn't find a valid docwhatever executable that option is looking for).

**Reasoning:**
This recent change breaks tests when docbook is not present, as the generated `docs/Makefile` tries to expand `DOCBOOK_TO_MAN` (which ends up empty) and tries to execute the xml directly when generating the manpage.

**Error output:**
```
[   50s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/expat-2.6.0/xmlwf'
[   50s] Making all in doc
[   50s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/expat-2.6.0/doc'
[   50s] rm -f xmlwf.1
[   50s] xmlwf.xml
[   50s] /usr/bin/bash: line 1: xmlwf.xml: command not found
[   50s] make[2]: *** [Makefile:606: xmlwf.1] Error 127
[   50s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/expat-2.6.0/doc'
[   50s] make[1]: *** [Makefile:650: all-recursive] Error 1
[   50s] make: *** [Makefile:509: all] Error 2
```

--- a/expat/configure.ac
+++ b/expat/configure.ac
@@ -357,11 +357,22 @@ AS_IF([test "x${DOCBOOK_TO_MAN}" != x -a "x$with_docbook" != xno],
   page for xmlwf.])])])
 
 dnl This will make sure that a release tarball shipping a pre-rendered xmlwf man page will
-dnl get it installed, independent of whether some flavor of docbook2man is available.
+dnl get it installed, when no working flavor of docbook2man is available (or wanted).
 dnl This relies on file xmlwf.1 being at least as recent as its source file xmlwf.xml.
 AS_IF([test -f "${srcdir}"/doc/xmlwf.1],
-  [AM_CONDITIONAL(WITH_DOCBOOK, [true])],
-  [AM_CONDITIONAL(WITH_DOCBOOK, [test "x${DOCBOOK_TO_MAN}" != x])])
+  [AM_CONDITIONAL(WITH_MANPAGE, [true])
+   AS_IF([test "x$with_docbook" = xno -o "x${DOCBOOK_TO_MAN}" = x],
+     [AM_CONDITIONAL(WITH_PREBUILT_MANPAGE, [true])
+      AM_CONDITIONAL(WITH_DISTRIBUTABLE_MANPAGE, [false])],
+     [AM_CONDITIONAL(WITH_PREBUILT_MANPAGE, [false])
+      AM_CONDITIONAL(WITH_DISTRIBUTABLE_MANPAGE, [true])])
+   ],
+  [AS_IF([test "x$with_docbook" != xno -a "x${DOCBOOK_TO_MAN}" != x],
+     [AM_CONDITIONAL(WITH_MANPAGE, [true])
+      AM_CONDITIONAL(WITH_DISTRIBUTABLE_MANPAGE, [true])],
+     [AM_CONDITIONAL(WITH_MANPAGE, [false])
+      AM_CONDITIONAL(WITH_DISTRIBUTABLE_MANPAGE, [false])])
+   AM_CONDITIONAL(WITH_PREBUILT_MANPAGE, [false])])
 
 dnl Configure CMake file templates
 dnl NOTE: The *_TRUE variables read here are Automake conditionals

--- a/expat/doc/Makefile.am
+++ b/expat/doc/Makefile.am
@@ -32,26 +32,24 @@
 
 .PHONY: dist-hook  # not inside conditional to avoid automake warning
 
-if WITH_DOCBOOK
+if WITH_MANPAGE
 dist_man_MANS = xmlwf.1
 
 xmlwf.1: xmlwf.xml
 	-rm -f $@
-	$(DOCBOOK_TO_MAN) $<
+	test x$(DOCBOOK_TO_MAN) != x && $(DOCBOOK_TO_MAN) $<
 	test -f $@ || mv XMLWF.1 $@
-else
+endif
+
+if !WITH_DISTRIBUTABLE_MANPAGE
 dist-hook:
 	@echo 'ERROR: Configure with --with-docbook for "make dist".' 1>&2
 	@false
 endif
 
-# https://www.gnu.org/software/automake/manual/automake.html#What-Gets-Cleaned
-.PHONY: clean-local
-clean-local: clean-local-check
-
-.PHONY: clean-local-check
-clean-local-check:
-	$(RM) xmlwf.1
+if !WITH_PREBUILT_MANPAGE
+CLEANFILES = xmlwf.1
+endif
 
 EXTRA_DIST = \
     ok.min.css \

